src/main/java/org/spongepowered/api/data/key/Keys.java

```diff
@@ -2097,6 +2097,13 @@
      */
     public static final Key<OptionalValue<UUID>> TAMED_OWNER = DummyObjectProvider.createExtendedFor(Key.class,"TAMED_OWNER");
 
+    /**
+     * Represents the {@link Key} for a targeted entity, like by a {@link ShulkerBullet}
+     *
+     * @see TargetedEntityData#value
+     */
+    public static final Key<OptionalValue<Entity>> TARGETED_ENTITY = DummyObjectProvider.createExtendedFor(Key.class,"TARGETED_ENTITY");
```

Comment:
Why is this an optional value? If there's no targeted entity, then the value shouldn't exist. Inversely, the value should only be returned as a valid value if there's a targeted entity (so you can support removing the targeted entity).